prelude/toolchains/cxx.bzl

```diff
@@ -170,50 +197,34 @@ def _system_cxx_toolchain_impl(ctx: AnalysisContext):
         CxxPlatformInfo(name = "x86_64"),
     ]
 
-def _windows_linker_wrapper(ctx: AnalysisContext, linker: cmd_args) -> cmd_args:
-    # Linkers pretty much all support @file.txt argument syntax to insert
-    # arguments from the given text file, usually formatted one argument per
-    # line.
-    #
-    # - GNU ld: https://gcc.gnu.org/onlinedocs/gcc/Overall-Options.html
-    # - lld is command line compatible with GNU ld
-    # - MSVC link.exe: https://learn.microsoft.com/en-us/cpp/build/reference/linking?view=msvc-170#link-command-files
-    #
-    # However, there is inconsistency in whether they support nesting of @file
-    # arguments inside of another @file.
-    #
-    # We wrap the linker to flatten @file arguments down to 1 level of nesting.
-    return cmd_script(
-        ctx = ctx,
-        name = "windows_linker",
-        cmd = cmd_args(
-            ctx.attrs.linker_wrapper[RunInfo],
-            linker,
-        ),
-        os = ScriptOs("windows"),
-    )
+def _run_info(args):
+    return None if args == None else RunInfo(args = [args])
 
 system_cxx_toolchain = rule(
     impl = _system_cxx_toolchain_impl,
     attrs = {
+        "toolchain_info": attrs.exec_dep(providers = [SystemCxxToolchainInfo], default = select({
```

Comment:
Thanks for the tip, this API was getting weird indeed. I created a separate rule called `cxx_tools_info_toolchain` (feel free to suggest a better name). I reused almost all code from the previous system_cxx_toolchain. Inclusively, for system_cxx_toolchain, I'm using a attrs.default_only of cxx_tools_info to provide the defaults rather than the previous way of using `default = ... if host_info().is_windows else ...` inside the rule declaration. This avoids repeating the defaults in two different places but let me know if you'd rather I separate these rules further.